getgauge/processor.py

```diff
@@ -166,16 +166,23 @@ def _get_impl_file_list(request, response, socket):
 
 
 def _get_stub_impl_content(request, response, socket):
-    response.messageType = Message.FileChanges
+    response.messageType = Message.FileDiff
     file_name = request.stubImplementationCodeRequest.implementationFilePath
     codes = request.stubImplementationCodeRequest.codes
-    response.fileChanges.fileName = file_name
+    response.fileDiff.filePath = file_name
+    code_concat = "\n".join(codes)
     existing_file_content = read_file_contents(file_name)
     if len(existing_file_content) > 0:
-        codes.insert(0, existing_file_content)
+        if (len(existing_file_content.strip().split("\n")) ==  len(existing_file_content.split("\n"))):
+            code_concat = "\n\n" + code_concat
+        else:
+            code_concat = "\n" + code_concat
```

Comment:
instead of join and adding strings, use `insert` method to add in code `list` and join at the end